agent/src/agent/destination/repository.py

```diff
@@ -28,5 +28,16 @@ def delete():
     session().commit()
 
 
+def get_auth_token(destination_: HttpDestination) -> Optional[AuthenticationToken]:
+    if not destination_.id:
```

Comment:
why is it possible?

Reply:
because we need to validate access key while creating destination, by that moment destination's not saved to db yet. I wasn't able to come up with a better solution

Reply:
hmm, as we'll use relationship it's not a problem anymore